src/main/java/appeng/me/cache/CraftingGridCache.java

```diff
@@ -445,7 +446,7 @@ public ImmutableCollection<ICraftingPatternDetails> getCraftingFor(
                     }
                 }
                 // no perfect match found, look for substitutions
-                if (details.canSubstitute() && !normalMode) {
+                if (details.canBeSubstitute() && !normalMode) {
```

Comment:
This is wrong, this search must only happen when "allow substitutions" is enabled